pkg/clusteragent/admission/mutate/agent_sidecar/util.go

```diff
@@ -0,0 +1,61 @@
+//go:build kubeapiserver
+
+package agentsidecar
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const agentSidecarContainerName = "datadog-agent-injected"
+
+func getDefaultSidecarTemplate() *corev1.Container {
+	agentContainer := &corev1.Container{
+		Env: []corev1.EnvVar{
```

Comment:
I don't see the `DD_FARGATE` envvar? 
is it plan to add it in a following pull request? 
IMO we can add it now, because for now the only environment we want to support is Fargate and without it, the agent is not behaving correctly as a side car container

Reply:
This is intentional here.

Based on my previous discussion with @levan-m , I understand that in order to keep the solution generic, we will start by having the `DD_EKS_FARGATE` env var set via the env overrides (which will be implemented in a future PR). So the user will have to set `DD_EKS_FARGATE` to `true` in the env overrides when running in fargate.

In the future, we can improve this by having an admission-controller-level config (like `admission.agent-sidecar.runtime` to set the environment on a global level. 

@levan-m any thoughts?

Reply:
Yes, we didn't include `DD_EKS_FARGATE` so the AC feature is generic and can be run e.g. on Kind. At the very least we will provide `DD_EKS_FARGATE` env var as an env override via Helm. If we have time we will add a runtime/provider setting and add the env var in binary.